ODH install using ArgoCD

martinpovolny opened the pull request:
based on @anishasthana 's demo

preview here: https://github.com/martinpovolny/continuous-deployment/blob/odh-install/docs/downstream/odh-install-crc.md

~to be rebased after https://github.com/operate-first/continuous-deployment/pull/33 is merged~

martinpovolny committed:
Add ODH instalation instructions.

martinpovolny commented:
Issues:

* the superset route lacks HTTPS

tumido commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+
+Go to https://argocd-server-aicoe-argocd-dev.apps-crc.testing/applications and click "New App". Enter these values:
+
+| Field            | Value                                                |
+|------------------|------------------------------------------------------|
+| Project:         | Default                                              |
+| Cluster:         | dev-cluster (https://api.crc.testing.6443)           |
+| Namespace:       | odh-operator                                         |
+| Repo URL:        | https://github.com/anishasthana/my-odh-install.git   |
+| Target revision: | HEAD                                                 |
+| Path:            | odh-operator/base                                    |
```
I think it would be better to provide the `Application` yaml snippet here, instead of a table of what user should click and type in the UI. What do you think?

Swap it for something like:

```
cat <<EOF | oc apply -f -
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: odh
spec:
  destination:
    namespace: odh-operator
    server: https://api.crc.testing.6443
  project: default
  source:
    path: odh-operator/base
    repoURL: https://github.com/anishasthana/my-odh-install.git
    targetRevision: HEAD
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
    syncOptions:
      - Validate=false
EOF
```

martinpovolny replied:
yes, it makes sense

martinpovolny replied:
But I'd rather put the resources into the repo than have them in-line in the doc. It will be easier to (automatically) test if things keep working. WDYT?

durandom replied:
+1 on the yaml snippet.

Put the yaml into the repo here in `/docs/downstream` for now. 
In the future I think the ODH specific steps should be in a separate repo anyways

tumido reviewed (changes requested):
I think providing the `Application` manifests directly would make it much easier to follow.

martinpovolny commented:
Also we should probably fork @anishasthana 's repo somewhere. Under operate-first probably?

And the instructions should probably mention people having to fork the repo and deploy from their own fork. WDYT?

anishasthana commented:
> Also we should probably fork @anishasthana 's repo somewhere. Under operate-first probably?
> 
> And the instructions should probably mention people having to fork the repo and deploy from their own fork. WDYT?

I agree, I think we can create a brand new repo somewhere within the operate-first umbrella.

durandom commented on docs/downstream/crc.md:
```diff
@@ -0,0 +1,34 @@
+# Deployment on CRC
+
+These is how to deploy ArgoCD on [CRC](https://developers.redhat.com/products/codeready-containers/overview).
```
`This is`...

martinpovolny replied:
Fixed in another PR.

durandom commented on examples/crc-create-myadmin.sh:
```diff
@@ -0,0 +1,8 @@
+oc create user myadmin
```
Let's move this to `docs/downstream` as well

HumairAK replied:
In that case I'd suggest putting this in a markdown, it would be weird having a script in a docs folder :)

martinpovolny replied:
I'd rather put it to `examples/`. But can create `examples/upstream/crc` or `examples/crc` WDTY?

It probably does not matter much at this point and we can figure it up as we add more content anyway.

durandom reviewed (approved):
I think we can merge this as is. 

Follow up

* recreate https://github.com/anishasthana/my-odh-install in our org
* exchange the table view and where to click in the UI with yaml files
* move the ODH specific parts to an ODH specific repo

HumairAK commented on docs/downstream/crc.md:
```diff
@@ -0,0 +1,34 @@
+
+These is how to deploy ArgoCD on [CRC](https://developers.redhat.com/products/codeready-containers/overview).
+
+## Instalation Steps
```
Installation*

martinpovolny replied:
fixing...

HumairAK commented on docs/downstream/crc.md:
```diff
@@ -0,0 +1,34 @@
+fix it with https://github.com/viaduct-ai/kustomize-sops/issues/60
+
+* As an alternative to installing the prerequisites locally, you can use Toolbox: https://github.com/containers/toolbox \
+   `toolbox create --image quay.io/aicoe/of-toolbox:v0.1.0` \
```
I would remove the instructions here on how to use the toolbox since they are mentioned in the link we'd need to make changes to multiple files if the versions change otherwise :)

martinpovolny replied:
This was part of another PR and is here because of the branching. Let's deal with it separately if needed.

HumairAK commented on docs/downstream/crc.md:
```diff
@@ -0,0 +1,34 @@
+
+## Instalation Steps
+
+ * Setup CRC https://developers.redhat.com/products/codeready-containers/overview
```
Can you convert all your links to hyperlinks  in the form [ text ] ( link )  it's a lot less cluttered this way imo

martinpovolny replied:
I do it for internal links. For external links I keep the URL in text. I personally like to see where I am heading when I click links. But it's probably a matter of preference.

HumairAK commented on docs/downstream/crc.md:
```diff
@@ -0,0 +1,34 @@
+
+ * Proceed with [setup_argocd_dev_environment.md](../setup_argocd_dev_environment.md)
+
+Your ArgoCD instance will be running at https://argocd-server-aicoe-argocd-dev.apps-crc.testing/
```
I would mention that they can login using openshift auth or as admin (user/pass: admin/admin)

martinpovolny replied:
+1

martinpovolny replied:
It was it the other PR, but adding it anyway.

HumairAK commented on docs/downstream/odh-install-crc.md:
````diff
@@ -0,0 +1,83 @@
+oc get packagemanifests -n openshift-marketplace | grep opendatahub-operator
+```
+
+If it's not present, then you need to troubleshoot the marketplace: https://github.com/operator-framework/operator-marketplace/issues/344
````
Can we convert these links to hyperlinks as well  in the form [ text ] ( link )

martinpovolny replied:
Same as above. I suggest doing it for relative links and keeping the external ones.

HumairAK commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+If it's not present, then you need to troubleshoot the marketplace: https://github.com/operator-framework/operator-marketplace/issues/344
+
+
+Then we need to create a projects named `odh-operator` and `odh-deployment`. (The name of the project is hard-coded in the `kustomize` files in https://github.com/anishasthana/my-odh-install and you can change is there using `kustomize`.)
```
> Then we need to create a projects named 
project * 

> https://github.com/anishasthana/my-odh-install 

This is @anishasthana 's personal repo, are we okay with linking there? is there a more official location? If not, then maybe we bring these files in to this org?

anishasthana replied:
I vote we create a new repo within the operatre first umbrella @HumairAK

durandom replied:
@martinpovolny I just copied it to https://github.com/operate-first/odh

HumairAK commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+
+Our ArgoCD SA `argocd-manager` needs to have access to the new project. On CRC I suggest giving `argocd-manager` access to all projects by creating a ClusterRoleBinding to a ClusterRole.
+
+`oc apply -f examples/argocd-cluster-binding.yaml`
```
I don't see this file?

martinpovolny replied:
added

HumairAK commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+
+## Creating the ArgoCD applications
+
+Now we can proceed with creating the "applications" in ArgoCD. We will create 2 applications:
```
I would rephrase this: 
> Now we can proceed with creating the "applications" in ArgoCD 

to say: 

> Now we can proceed with creating the ArgoCD Applications.

martinpovolny replied:
+1

HumairAK commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+Now we can proceed with creating the "applications" in ArgoCD. We will create 2 applications:
+
+  1. For the ODH operator. (Installs the ODH operator itself `odh-operator`.)
+  1. For the ODH deployment. (Installs ODH components into `odh-deployment`.)
```
Remove "For" and just say "The ODH operator" and "The ODH deployment"

martinpovolny replied:
+1

HumairAK commented on docs/downstream/odh-install-crc.md:
```diff
@@ -0,0 +1,83 @@
+
+
+Create the app and you will see Argo deploying resources:
+(doc/assets/images/upstream/odh-operator-argocd.png)
```
this link is invalid, I'm guessing it's because it should be "docs" not "doc"

martinpovolny replied:
thx, it was a line I forgot to remove. the correct URL is one line down

martinpovolny committed:
Minor fixes in ArgoCD CRC guide.

sesheta commented:
<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Failed
- hook id: remove-tabs
- exit code: 1
- files were modified by this hook

Substituting tabs in: docs/downstream/crc.md by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

manifests/overlays/dev/applications/dev_app/dev-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  16:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/dev.cluster.example.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  12:81     error    line too long (227 > 80 characters)  (line-length)
  21:81     error    line too long (288 > 80 characters)  (line-length)
  23:8      error    too many spaces after hyphen  (hyphens)
  23:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/projects/projects.yaml
  1:1       warning  missing document start "---"  (document-start)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  10:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/services/argocd_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/crds/aggregate-to-admin.yaml
  1:1       warning  missing document start "---"  (document-start)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  14:3      error    wrong indentation: expected 4 but found 2  (indentation)
  30:1      error    wrong indentation: expected 2 but found 0  (indentation)
  31:3      error    wrong indentation: expected 4 but found 2  (indentation)
  33:3      error    wrong indentation: expected 4 but found 2  (indentation)
  35:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/secrets/auth/argocd-secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:81      error    line too long (207 > 80 characters)  (line-length)
  10:81     error    line too long (160 > 80 characters)  (line-length)
  11:81     error    line too long (1432 > 80 characters)  (line-length)
  18:81     error    line too long (288 > 80 characters)  (line-length)
  20:8      error    too many spaces after hyphen  (hyphens)
  20:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/rbac/argocd-manager-role.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  13:3      error    wrong indentation: expected 4 but found 2  (indentation)
  15:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/serviceaccounts/argocd-dex-server.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:81     error    line too long (94 > 80 characters)  (line-length)

manifests/base/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/serviceaccounts/argocd-manager.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/projects/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/services/argocd_server_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-cluster-binding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role-binding.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  5:1       error    wrong indentation: expected 2 but found 0  (indentation)
  11:1      error    wrong indentation: expected 2 but found 0  (indentation)
  25:1      error    wrong indentation: expected 2 but found 0  (indentation)
  29:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/routes/argocd_server_route.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/base/deployments/argocd-repo-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  12:9      error    wrong indentation: expected 10 but found 8  (indentation)
  15:9      error    wrong indentation: expected 10 but found 8  (indentation)
  22:7      error    wrong indentation: expected 8 but found 6  (indentation)

manifests/overlays/dev/applications/dev_app/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/gpg/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

.aicoe-ci.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-manager-rolebinding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/secrets/clusters/.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/base/deployments/argocd-redis_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  19:7      error    wrong indentation: expected 8 but found 6  (indentation)
  20:9      error    wrong indentation: expected 10 but found 8  (indentation)
  28:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/crds/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/clusters/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/dex-server-sa.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  15:81     error    line too long (288 > 80 characters)  (line-length)
  17:8      error    too many spaces after hyphen  (hyphens)
  17:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/base/deployments/argocd-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  11:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/overlays/dev/secrets/gpg/secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:81      error    line too long (6448 > 80 characters)  (line-length)
  13:81     error    line too long (288 > 80 characters)  (line-length)
  15:8      error    too many spaces after hyphen  (hyphens)
  15:5      error    wrong indentation: expected 8 but found 4  (indentation)
```
</details>

sesheta commented:
<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Failed
- hook id: remove-tabs
- exit code: 1
- files were modified by this hook

Substituting tabs in: docs/downstream/crc.md by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

manifests/overlays/dev/applications/dev_app/dev-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  16:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/dev.cluster.example.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  12:81     error    line too long (227 > 80 characters)  (line-length)
  21:81     error    line too long (288 > 80 characters)  (line-length)
  23:8      error    too many spaces after hyphen  (hyphens)
  23:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/projects/projects.yaml
  1:1       warning  missing document start "---"  (document-start)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  10:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/services/argocd_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/crds/aggregate-to-admin.yaml
  1:1       warning  missing document start "---"  (document-start)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  14:3      error    wrong indentation: expected 4 but found 2  (indentation)
  30:1      error    wrong indentation: expected 2 but found 0  (indentation)
  31:3      error    wrong indentation: expected 4 but found 2  (indentation)
  33:3      error    wrong indentation: expected 4 but found 2  (indentation)
  35:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/secrets/auth/argocd-secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:81      error    line too long (207 > 80 characters)  (line-length)
  10:81     error    line too long (160 > 80 characters)  (line-length)
  11:81     error    line too long (1432 > 80 characters)  (line-length)
  18:81     error    line too long (288 > 80 characters)  (line-length)
  20:8      error    too many spaces after hyphen  (hyphens)
  20:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/rbac/argocd-manager-role.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  13:3      error    wrong indentation: expected 4 but found 2  (indentation)
  15:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/serviceaccounts/argocd-dex-server.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:81     error    line too long (94 > 80 characters)  (line-length)

manifests/base/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/serviceaccounts/argocd-manager.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/projects/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/services/argocd_server_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-cluster-binding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role-binding.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  5:1       error    wrong indentation: expected 2 but found 0  (indentation)
  11:1      error    wrong indentation: expected 2 but found 0  (indentation)
  25:1      error    wrong indentation: expected 2 but found 0  (indentation)
  29:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/routes/argocd_server_route.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/base/deployments/argocd-repo-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  12:9      error    wrong indentation: expected 10 but found 8  (indentation)
  15:9      error    wrong indentation: expected 10 but found 8  (indentation)
  22:7      error    wrong indentation: expected 8 but found 6  (indentation)

manifests/overlays/dev/applications/dev_app/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/gpg/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

.aicoe-ci.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-manager-rolebinding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/secrets/clusters/.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/base/deployments/argocd-redis_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  19:7      error    wrong indentation: expected 8 but found 6  (indentation)
  20:9      error    wrong indentation: expected 10 but found 8  (indentation)
  28:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/crds/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/clusters/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/dex-server-sa.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  15:81     error    line too long (288 > 80 characters)  (line-length)
  17:8      error    too many spaces after hyphen  (hyphens)
  17:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/base/deployments/argocd-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  11:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/overlays/dev/secrets/gpg/secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:81      error    line too long (6448 > 80 characters)  (line-length)
  13:81     error    line too long (288 > 80 characters)  (line-length)
  15:8      error    too many spaces after hyphen  (hyphens)
  15:5      error    wrong indentation: expected 8 but found 4  (indentation)
```
</details>

martinpovolny committed:
Add ODH instalation instructions: more updates and fixes.

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/continuous-deployment/pull/34#pullrequestreview-494765093" title="Approved">durandom</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fcontinuous-deployment).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/continuous-deployment/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

martinpovolny commented:
> recreate https://github.com/anishasthana/my-odh-install in our org

done by you, I updated the links

> exchange the table view and where to click in the UI with yaml files

I have added links to the files and the files while keeping the tables. I think that it helps to understand what is happening to see the tables.

sesheta commented:
<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Failed
- hook id: remove-tabs
- exit code: 1
- files were modified by this hook

Substituting tabs in: docs/downstream/crc.md by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

manifests/overlays/dev/serviceaccounts/argocd-dex-server.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:81     error    line too long (94 > 80 characters)  (line-length)

manifests/base/routes/argocd_server_route.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/dev_app/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/deployments/argocd-repo-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  12:9      error    wrong indentation: expected 10 but found 8  (indentation)
  15:9      error    wrong indentation: expected 10 but found 8  (indentation)
  22:7      error    wrong indentation: expected 8 but found 6  (indentation)

manifests/base/services/argocd_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  13:3      error    wrong indentation: expected 4 but found 2  (indentation)
  15:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/services/argocd_server_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/crds/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/clusters/dev.cluster.example.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  12:81     error    line too long (227 > 80 characters)  (line-length)
  21:81     error    line too long (288 > 80 characters)  (line-length)
  23:8      error    too many spaces after hyphen  (hyphens)
  23:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/projects/projects.yaml
  1:1       warning  missing document start "---"  (document-start)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  10:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/serviceaccounts/argocd-manager.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/projects/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

examples/argocd-manager-rolebinding.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-cluster-binding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role-binding.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/deployments/argocd-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  11:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/base/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  5:1       error    wrong indentation: expected 2 but found 0  (indentation)
  11:1      error    wrong indentation: expected 2 but found 0  (indentation)
  25:1      error    wrong indentation: expected 2 but found 0  (indentation)
  29:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/applications/dev_app/dev-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  16:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/gpg/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

.aicoe-ci.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/odh-deployment-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  17:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/odh-operator-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  17:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/crds/aggregate-to-admin.yaml
  1:1       warning  missing document start "---"  (document-start)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  14:3      error    wrong indentation: expected 4 but found 2  (indentation)
  30:1      error    wrong indentation: expected 2 but found 0  (indentation)
  31:3      error    wrong indentation: expected 4 but found 2  (indentation)
  33:3      error    wrong indentation: expected 4 but found 2  (indentation)
  35:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/secrets/auth/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/dex-server-sa.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  15:81     error    line too long (288 > 80 characters)  (line-length)
  17:8      error    too many spaces after hyphen  (hyphens)
  17:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/secrets/auth/argocd-secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:81      error    line too long (207 > 80 characters)  (line-length)
  10:81     error    line too long (160 > 80 characters)  (line-length)
  11:81     error    line too long (1432 > 80 characters)  (line-length)
  18:81     error    line too long (288 > 80 characters)  (line-length)
  20:8      error    too many spaces after hyphen  (hyphens)
  20:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/base/deployments/argocd-redis_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  19:7      error    wrong indentation: expected 8 but found 6  (indentation)
  20:9      error    wrong indentation: expected 10 but found 8  (indentation)
  28:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/overlays/dev/secrets/gpg/secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:81      error    line too long (6448 > 80 characters)  (line-length)
  13:81     error    line too long (288 > 80 characters)  (line-length)
  15:8      error    too many spaces after hyphen  (hyphens)
  15:5      error    wrong indentation: expected 8 but found 4  (indentation)
```
</details>

sesheta commented:
<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Failed
- hook id: remove-tabs
- exit code: 1
- files were modified by this hook

Substituting tabs in: docs/downstream/crc.md by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

manifests/overlays/dev/serviceaccounts/argocd-dex-server.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:81     error    line too long (94 > 80 characters)  (line-length)

manifests/base/routes/argocd_server_route.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/dev_app/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/deployments/argocd-repo-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  12:9      error    wrong indentation: expected 10 but found 8  (indentation)
  15:9      error    wrong indentation: expected 10 but found 8  (indentation)
  22:7      error    wrong indentation: expected 8 but found 6  (indentation)

manifests/base/services/argocd_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  13:3      error    wrong indentation: expected 4 but found 2  (indentation)
  15:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/services/argocd_server_metrics_service_patch.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/crds/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/clusters/dev.cluster.example.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  12:81     error    line too long (227 > 80 characters)  (line-length)
  21:81     error    line too long (288 > 80 characters)  (line-length)
  23:8      error    too many spaces after hyphen  (hyphens)
  23:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/projects/projects.yaml
  1:1       warning  missing document start "---"  (document-start)
  7:3       error    wrong indentation: expected 4 but found 2  (indentation)
  10:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/base/serviceaccounts/argocd-manager.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/projects/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

examples/argocd-manager-rolebinding.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/argocd-cluster-binding.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/rbac/argocd-manager-role-binding.yaml
  1:1       warning  missing document start "---"  (document-start)
  10:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/base/deployments/argocd-server_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:7       error    wrong indentation: expected 8 but found 6  (indentation)
  11:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/base/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  5:1       error    wrong indentation: expected 2 but found 0  (indentation)
  11:1      error    wrong indentation: expected 2 but found 0  (indentation)
  25:1      error    wrong indentation: expected 2 but found 0  (indentation)
  29:1      error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/applications/dev_app/dev-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  16:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/gpg/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

.aicoe-ci.yaml
  1:1       warning  missing document start "---"  (document-start)

manifests/overlays/dev/applications/kustomization.yaml
  1:1       warning  missing document start "---"  (document-start)
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)

.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/odh-deployment-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  17:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/overlays/dev/secrets/clusters/.sops.yaml
  1:1       warning  missing document start "---"  (document-start)

examples/odh-operator-app.yaml
  1:1       warning  missing document start "---"  (document-start)
  17:5      error    wrong indentation: expected 6 but found 4  (indentation)

manifests/crds/aggregate-to-admin.yaml
  1:1       warning  missing document start "---"  (document-start)
  8:1       error    wrong indentation: expected 2 but found 0  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)
  11:3      error    wrong indentation: expected 4 but found 2  (indentation)
  14:3      error    wrong indentation: expected 4 but found 2  (indentation)
  30:1      error    wrong indentation: expected 2 but found 0  (indentation)
  31:3      error    wrong indentation: expected 4 but found 2  (indentation)
  33:3      error    wrong indentation: expected 4 but found 2  (indentation)
  35:3      error    wrong indentation: expected 4 but found 2  (indentation)

manifests/overlays/dev/secrets/auth/secret-generator.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:1       error    wrong indentation: expected 2 but found 0  (indentation)

manifests/overlays/dev/secrets/auth/dex-server-sa.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  15:81     error    line too long (288 > 80 characters)  (line-length)
  17:8      error    too many spaces after hyphen  (hyphens)
  17:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/overlays/dev/secrets/auth/argocd-secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  9:81      error    line too long (207 > 80 characters)  (line-length)
  10:81     error    line too long (160 > 80 characters)  (line-length)
  11:81     error    line too long (1432 > 80 characters)  (line-length)
  18:81     error    line too long (288 > 80 characters)  (line-length)
  20:8      error    too many spaces after hyphen  (hyphens)
  20:5      error    wrong indentation: expected 8 but found 4  (indentation)

manifests/base/deployments/argocd-redis_patch.yaml
  1:1       warning  missing document start "---"  (document-start)
  19:7      error    wrong indentation: expected 8 but found 6  (indentation)
  20:9      error    wrong indentation: expected 10 but found 8  (indentation)
  28:9      error    wrong indentation: expected 10 but found 8  (indentation)

manifests/overlays/dev/secrets/gpg/secret.enc.yaml
  1:1       warning  missing document start "---"  (document-start)
  6:81      error    line too long (6448 > 80 characters)  (line-length)
  13:81     error    line too long (288 > 80 characters)  (line-length)
  15:8      error    too many spaces after hyphen  (hyphens)
  15:5      error    wrong indentation: expected 8 but found 4  (indentation)
```
</details>

martinpovolny commented:
@tumido : Can you merge this one now or do you have other comments?

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/continuous-deployment/pull/34#pullrequestreview-494765093" title="Approved">durandom</a>*, *<a href="https://github.com/operate-first/continuous-deployment/pull/34#pullrequestreview-499473727" title="Approved">tumido</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fcontinuous-deployment).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/continuous-deployment/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
